From f246a4ddb58ef07159bed97cdab9e078e2cc92f5 Mon Sep 17 00:00:00 2001
From: Joseph Pirozzo <pirozzoj@google.com>
Date: Mon, 25 May 2020 10:36:26 -0700
Subject: [PATCH 1/3] Enable bitpool sanity checks

Enable bitpool sanity checks to run all the time, not just in debug
mode.

Tag: #security
Test: sbcdecoder_fuzzer
Bug: 146398979
Change-Id: Iff58305cd18de35e37290f0c09fba01ee14e787a
(cherry picked from commit 59c234a8fddda37147bb3fe1dd3b3a668828bcab)
---
 embdrv/sbc/decoder/srce/decoder-sbc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/embdrv/sbc/decoder/srce/decoder-sbc.c b/embdrv/sbc/decoder/srce/decoder-sbc.c
index 20c5b67b7..674645e27 100644
--- a/embdrv/sbc/decoder/srce/decoder-sbc.c
+++ b/embdrv/sbc/decoder/srce/decoder-sbc.c
@@ -311,7 +311,6 @@ OI_STATUS OI_CODEC_SBC_DecodeFrame(OI_CODEC_SBC_DECODER_CONTEXT* context,
     return OI_CODEC_SBC_CHECKSUM_MISMATCH;
   }
 
-#ifdef OI_DEBUG
   /*
    * Make sure the bitpool values are sane.
    */
@@ -328,7 +327,6 @@ OI_STATUS OI_CODEC_SBC_DecodeFrame(OI_CODEC_SBC_DECODER_CONTEXT* context,
            OI_SBC_MaxBitpool(&context->common.frameInfo)));
     return OI_STATUS_INVALID_PARAMETERS;
   }
-#endif
 
   /*
    * Now decode the SBC data. Partial decode is not yet implemented for an SBC
-- 
2.17.1

